lib/auth/auth.go

```diff
@@ -3686,6 +3686,37 @@ func (a *Server) deleteMFADeviceSafely(ctx context.Context, user, deviceName str
 		return nil, trace.BadParameter("unexpected second factor type: %s", sf)
 	}
 
+	canDeleteLastPasskey := func() (bool, error) {
+		if !authPref.GetAllowPasswordless() || numResidentKeys > 1 || !isResidentKey(deviceToDelete) {
+			return true, nil
+		}
+
+		u, err := a.Services.GetUser(ctx, user, false /* withSecrets */)
```

Comment:
```suggestion
		// Deleting the last passkey is OK if the user has a password set and an
		// additional MFA device, otherwise they would be locked out.
		u, err := a.Services.GetUser(ctx, user, false /* withSecrets */)
```